framework/wazuh/core/mitre.py

```diff
@@ -186,6 +190,17 @@ def _execute_data_query(self):
                                         zip(references['items'], references_no_id['items']) if
                                         row['id'] == mitigation['id']]
 
+            # Take reference with external_id checking it is not None
+            reference_external_id = next(
+                (row_no_id for row_no_id in mitigation['references'] if
+                 'external_id' in row_no_id and row_no_id['external_id']), {})
+            if 'description' in reference_external_id:
+                reference_external_id.pop('description')
+            # Delete the reference from references and update the object
+            if reference_external_id in mitigation['references']:
```

Comment:
```suggestion
            if reference_external_id:
```
If the `reference_external_id` is not empty it is guaranteed to be in `mitigation['references']`